### PR TITLE
Add tiny-mem

### PR DESCRIPTION
**Pull request by gabmontes**
This PR adds the `tiny-mem` package Tests and documentation were added too.

### Process checklist

- [x] Manual tests passed or not applicable.
- [x] Automated tests added or not applicable.
- [x] Documentation updated or not required.

### Metrics

Actual effort: 1.5h

### CODE REVIEW

**Review by gndelia — CHANGES_REQUESTED** [review]

File: packages/tiny-mem/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      "src/**/*.js"
+    ],
+    "reporter": [
+      "lcov",
+      "text"
+    ]
+  },
+  "engines": {
+    "node": ">= 14"
+  }
```
Comment (gndelia) [inline packages/tiny-mem/package.json:34]: 14 is not even on maintenance mode... Should we use 16 right from scratch? This also applies for other packages..

File: packages/tiny-mem/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+const memoized = memoize(seq)
+
+memoized('one') // Returns 1
+
+memoized('two') // Returns 2
+memoized('two') // Returns 2 again
+
+memoized('one') // Returns 1 again
+
+memoized('two', 'one') // Still returns 2
```
[the remark this answers is not in the document]
Comment (gabmontes) [reply packages/tiny-mem/README.md:35]: Yeah, better to stick to known implementations as `tiny-mem` could be a drop-in replacement.

File: packages/tiny-mem/package.json (hunk 1 above)
Comment (gabmontes) [reply packages/tiny-mem/package.json:34]: The code is compatible with Node 14 and up. If someone is using that old version, so be it. We should not prevent them from using this package.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,21 @@ This is a collection of small and useful packages we use across many of our solu
...
+### Promises
+
+- [`promise-mem`](./packages/promise-mem): Memoizes a promise-returning function.
+- [`promise-stale-if-error`](./packages/promise-stale-if-error): Wraps a promise-returning function so it resolves to the last good value on error.
+- [`promise-swr`](./packages/promise-swr): Caches a promise-returning function with a stale-while-revalidate strategy.
+- [`promise-throttle-bucket`](./packages/promise-throttle-bucket): Throttles a promise-returning function using the token-bucket algorithm.
+
+### Timy ones
+
+- [`tiny-mem`](./oackages/tiny-mem): Minimal, no-dependencies memoization helper.
```
[the remark this answers is not in the document]
Comment (gabmontes) [reply README.md:16]: Fixed!

**Review by mbonfoster — APPROVED** [review]

**Review by maxivarela — APPROVED** [review]

**Pull request merged**